src/coreclr/gc/gcpriv.h

```diff
@@ -3364,7 +3377,7 @@ class gc_heap
     PER_HEAP
     int pinning_seg_interval;
     PER_HEAP
-    int num_gen0_segs;
+    int num_gen0_regions;
```

Comment:
It looks like this could be a local variable instead of a field.

Reply:
I specifically wanted this to be a global so it introduces some variation instead of always pinning the 1st, 3rd, etc gen0 region. but I changed it to size_t instead of int.